### DISCUSSION TITLE
Hardly/Not converged Picard iteration

### DISCUSSION THREAD

**Discussion by Bearunner** [idaholab/moose#26697] [Q&A Modules: General]
Dear all,

I am using the [MultiApps], [Transfers] and Picard interation to simulate the 1D pipe flow and 3D porous media flow (i.e., porous_flow).

I set the related information in [Executioner], [MultiApps] and [Transfers].  But the Picard iteration is hardly or not converged as shown in the following log. Even if I increase the fixed_point_rel_tol and fixed_point_abs_tol to 1e-5. It still the same.

```
**[Executioner]**
  type = Transient
  start_time =  ${starting_time}
  end_time = ${end_time}
  dtmax = ${1_day}
  l_tol = 1e-8
  nl_rel_tol = 1e-8
  nl_abs_tol = 1e-8
  l_max_its = 100
  nl_max_its = 100
  solve_type = NEWTON
  fixed_point_rel_tol = 1e-8
  fixed_point_max_its = 100
  fixed_point_abs_tol = 1e-8
  disable_fixed_point_residual_norm_check = true
  relaxation_factor = 0.1
 []

**[MultiApps]**
  [./pipe]
    type = TransientMultiApp
    input_files = pipe.i
    positions = '0 -97.5 0'
    library_path = /home/projects/pipe/lib
    app_type = pipeApp
    catch_up = true
  [../]
[]

**[Transfers]**
  [./to_pipe]
    type = MultiAppGeneralFieldShapeEvaluationTransfer
    to_multi_app = pipe
    source_variable = porepressure
    variable = pp_res
    from_blocks = a
    to_blocks = b
    greedy_search = true
    use_nearest_app = true
    error_on_miss = true
  [../]
  [./from_pipe]
    type = MultiAppGeneralFieldNearestNodeTransfer
    from_multi_app = pipe
    source_variable = p
    variable = pp_pipe
    from_blocks = c
    to_blocks = d
  [../]
[]
```
**Part of the log:**
```
[36mpipe_sub_inj0: [39m
[36mpipe_sub_inj0: [39mTime Step 30, time = 3227.17, dt = 646.235
[36mpipe_sub_inj0: [39m 0 Nonlinear |R| = [32m1.393859e-01[39m
[36mpipe_sub_inj0: [39m 1 Nonlinear |R| = [32m1.626299e-04[39m
[36mpipe_sub_inj0: [39m 2 Nonlinear |R| = [32m1.105107e-04[39m
[36mpipe_sub_inj0: [39m 3 Nonlinear |R| = [32m1.040351e-07[39m
[36mpipe_sub_inj0: [39m 4 Nonlinear |R| = [32m1.076632e-09[39m
[36mpipe_sub_inj0: [39m[32m Solve Converged![39m

[35mFixed point residual norm after TIMESTEP_BEGIN MultiApps: [32m1.405005e-08[39m
[35m
Main app solve:[39m
 0 Nonlinear |R| = [32m1.405005e-08[39m
 1 Nonlinear |R| = [32m4.987454e-11[39m
[32m Solve Converged![39m

 0 Picard |R| = [32m1.591909e+00[39m
 1 Picard |R| = [33m1.591663e+00[39m
 2 Picard |R| = [32m4.275689e-01[39m
 3 Picard |R| = [32m2.974149e-01[39m
 4 Picard |R| = [32m2.178361e-01[39m
 5 Picard |R| = [32m1.633035e-01[39m
 6 Picard |R| = [32m1.236165e-01[39m
 7 Picard |R| = [32m9.395439e-02[39m
 8 Picard |R| = [32m7.153569e-02[39m
 9 Picard |R| = [32m5.450221e-02[39m
10 Picard |R| = [32m4.154714e-02[39m
11 Picard |R| = [32m3.167794e-02[39m
12 Picard |R| = [32m2.415499e-02[39m
13 Picard |R| = [32m1.842036e-02[39m
14 Picard |R| = [32m1.404766e-02[39m
15 Picard |R| = [32m1.071319e-02[39m
16 Picard |R| = [32m8.201140e-03[39m
17 Picard |R| = [32m6.254511e-03[39m
18 Picard |R| = [32m4.769999e-03[39m
19 Picard |R| = [32m3.637849e-03[39m
20 Picard |R| = [32m2.774418e-03[39m
21 Picard |R| = [32m2.115921e-03[39m
22 Picard |R| = [32m1.613717e-03[39m
23 Picard |R| = [32m1.230709e-03[39m
24 Picard |R| = [32m9.386061e-04[39m
25 Picard |R| = [32m7.158326e-04[39m
26 Picard |R| = [32m5.459334e-04[39m
27 Picard |R| = [32m4.163589e-04[39m
28 Picard |R| = [32m3.175383e-04[39m
29 Picard |R| = [32m2.421722e-04[39m
30 Picard |R| = [32m1.846938e-04[39m
31 Picard |R| = [32m1.408577e-04[39m
32 Picard |R| = [32m1.074259e-04[39m
33 Picard |R| = [32m8.192889e-05[39m
34 Picard |R| = [32m6.248350e-05[39m
35 Picard |R| = [32m4.765336e-05[39m
36 Picard |R| = [32m3.634308e-05[39m
37 Picard |R| = [32m2.771724e-05[39m
38 Picard |R| = [32m2.113879e-05[39m
39 Picard |R| = [32m1.612162e-05[39m
40 Picard |R| = [32m1.229524e-05[39m
41 Picard |R| = [32m9.377037e-06[39m
42 Picard |R| = [32m7.151450e-06[39m
43 Picard |R| = [32m5.454094e-06[39m
44 Picard |R| = [32m4.159596e-06[39m
45 Picard |R| = [32m3.172339e-06[39m
46 Picard |R| = [32m2.419403e-06[39m
47 Picard |R| = [32m1.844672e-06[39m
48 Picard |R| = [32m1.406827e-06[39m
49 Picard |R| = [32m1.072903e-06[39m
50 Picard |R| = [32m8.182389e-07[39m
51 Picard |R| = [32m6.240221e-07[39m
52 Picard |R| = [32m4.759045e-07[39m
53 Picard |R| = [32m3.629440e-07[39m
54 Picard |R| = [32m2.767958e-07[39m
55 Picard |R| = [32m2.110956e-07[39m
56 Picard |R| = [32m1.609900e-07[39m
57 Picard |R| = [32m1.227775e-07[39m
58 Picard |R| = [32m9.363509e-08[39m
59 Picard |R| = [32m7.140995e-08[39m
60 Picard |R| = [32m5.446004e-08[39m
61 Picard |R| = [32m4.153347e-08[39m
62 Picard |R| = [32m3.167511e-08[39m
63 Picard |R| = [32m2.415671e-08[39m
64 Picard |R| = [32m1.842289e-08[39m
65 Picard |R| = [32m1.405005e-08[39m
```

I also follow the first and third tips in https://github.com/idaholab/moose/discussions/23102. It still the same.

>1. Set All execute_on to default (including which of [MultiApps] block). 
> 2. Set disable_fixed_point_residual_norm_check=true , use custom_pp and relax the convergence criteria a little
> 3. Set relaxation_factor of main app (of [Executioner] block) to extremely small (around 0.1)

Could you please help me on this? Thanks a lot.

B

**Comment by GiudGiud** [comment]
Hello

Setting the   relaxation_factor to  0.1 is meant to dampen the iteration. It will help getting a solution if the coupling is not stable, but it will slow down convergence.

What results do you get without relaxation?
```
65 Picard |R| = �[32m1.405005e-08�[39m
```
this looks to be converging. Just slowly

Guillaume

**Reply by Bearunner** [reply]
Hello, Guillaume. Thanks for your reply.

There is no obvious difference when I don't set the 'relaxation_factor'.

With time, the picard iteration needs more than 100 times to converge. The dt is still in low value, not greatly increasing.

**Reply by GiudGiud** [reply]
I think you need to specify which variables to transform to use relaxation btw. Use the transformed_variables parameter to do so

**Reply by Bearunner** [reply]
Thanks for the tip. 

If I add transformed_variables = 'porepressure' ' and don't add the 'relaxation_factor= 0.1', there is no obvious difference to the mentioned hardly converged Picard iteration. 

If I set the 'relaxation_factor= 0.1' and 'transformed_variables = 'porepressure'' together, it cannot converged at 2nd step. 

pipe:
pipe: Time Step 2, time = 2.25, dt = 1.25
pipe:  0 Nonlinear |R| = 5.005947e+01
pipe:  1 Nonlinear |R| = 5.602494e-01
pipe:  2 Nonlinear |R| = 2.560555e-04
pipe:  3 Nonlinear |R| = 2.643941e-07

pipe:  Solve Converged!

Fixed point residual norm after TIMESTEP_BEGIN MultiApps: 2.051577e-03

Main app solve:
 0 Nonlinear |R| = 2.051577e-03
 1 Nonlinear |R| = 2.887538e-06
 2 Nonlinear |R| = 4.083398e-08
 3 Nonlinear |R| = 9.518876e-10
 Solve Converged!

 0 Picard |R| = 2.756204e+00
 1 Picard |R| = 9.648449e+00
 2 Picard |R| = 1.006712e+00
 3 Picard |R| = 9.097428e-01
 4 Picard |R| = 8.283195e-01
 5 Picard |R| = 7.541837e-01
 6 Picard |R| = 6.866832e-01
 7 Picard |R| = 6.252241e-01
 8 Picard |R| = 5.692657e-01
 9 Picard |R| = 5.183156e-01
10 Picard |R| = 4.719257e-01
11 Picard |R| = 4.296877e-01
12 Picard |R| = 3.912301e-01
13 Picard |R| = 3.562146e-01
14 Picard |R| = 3.243329e-01
15 Picard |R| = 2.953047e-01
16 Picard |R| = 2.688746e-01
17 Picard |R| = 2.448100e-01
18 Picard |R| = 2.228992e-01
19 Picard |R| = 2.029495e-01
20 Picard |R| = 1.847853e-01
21 Picard |R| = 1.682468e-01
22 Picard |R| = 1.531886e-01
23 Picard |R| = 1.394780e-01
24 Picard |R| = 1.269946e-01
25 Picard |R| = 1.156285e-01
26 Picard |R| = 1.052796e-01
27 Picard |R| = 9.585697e-02
28 Picard |R| = 8.727768e-02
29 Picard |R| = 7.946625e-02
30 Picard |R| = 7.235394e-02
31 Picard |R| = 6.587820e-02
32 Picard |R| = 5.998204e-02
33 Picard |R| = 5.461360e-02
34 Picard |R| = 4.972564e-02
35 Picard |R| = 4.527515e-02
36 Picard |R| = 4.122299e-02
37 Picard |R| = 3.753349e-02
38 Picard |R| = 3.417422e-02
39 Picard |R| = 3.111560e-02
40 Picard |R| = 2.833073e-02
41 Picard |R| = 2.579511e-02
42 Picard |R| = 2.348642e-02
43 Picard |R| = 2.138437e-02
44 Picard |R| = 1.947046e-02
45 Picard |R| = 1.772784e-02
46 Picard |R| = 1.614118e-02
47 Picard |R| = 1.469653e-02
48 Picard |R| = 1.338118e-02
49 Picard |R| = 1.218356e-02
50 Picard |R| = 1.109312e-02
51 Picard |R| = 1.010028e-02
52 Picard |R| = 9.196299e-03
53 Picard |R| = 8.373225e-03
54 Picard |R| = 7.623816e-03
55 Picard |R| = 6.941479e-03
56 Picard |R| = 6.320213e-03
57 Picard |R| = 5.754550e-03
58 Picard |R| = 5.239514e-03
59 Picard |R| = 4.770575e-03
60 Picard |R| = 4.343605e-03
61 Picard |R| = 3.954850e-03
62 Picard |R| = 3.600889e-03
63 Picard |R| = 3.278607e-03
64 Picard |R| = 2.985170e-03
65 Picard |R| = 2.717996e-03
66 Picard |R| = 2.474734e-03
67 Picard |R| = 2.253244e-03
68 Picard |R| = 2.051577e-03

**Reply by GiudGiud** [reply]
> If I add transformed_variables = 'porepressure' ' and don't add the 'relaxation_factor= 0.1', there is no obvious difference to the mentioned hardly converged Picard iteration.

this is expected. There's no transformation if the relaxation factor is not specified.

> If I set the 'relaxation_factor= 0.1' and 'transformed_variables = 'porepressure'' together, it cannot converged at 2nd step.

it seems to be converging, just very slowly.
Could you solve the multiapp and the main app in the same simulation? Right now you use Picard to solve the coupled problem. If they are in the same app you can use Newton or PJFNK

**Reply by Bearunner** [reply]
Actually, we initially  solve the multiapp and the main app in the same simulation by merging 1D pipe flow (N-S eqations with pressure, flowrate （and temperature） as main variables) and 3D porous media flow (similar to porous_flow module with pressure （and temperature） as main variables) into one app. 

we used the  Newton or PJFNK and tries the following [Preconditioning]s which were working well in standalone two apps seperately. However, all of them cannot converged. I guess the the [Preconditioning] cannot working well in this mixture-flow simulation. Could you please help me on this? Thanks.

**3D porous media flow：**
[Preconditioning]
  [basic]
    type = SMP
    full = true
    petsc_options = '-ksp_diagonal_scale -ksp_diagonal_scale_fix'
    petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -pc_asm_overlap'
    petsc_options_value = ' asm      lu           NONZERO                   2'
  []
[]

**1D pipe flow：**

[Preconditioning]
[./basic]
  type = SMP
  full = true
  petsc_options_iname = '-pc_type -sub_pc_type -sub_pc_factor_shift_type -snes_type -snes_linesearch_type'
  petsc_options_value = ' bjacobi  ilu          NONZERO                   newtonls   basic               '
[../]
[]


 0 Nonlinear |R| = 7.216034e+00
    |residual|_2 of individual variables:
                          q: 3.34388
                          p: 6.3945
      0 Linear |R| = 7.216034e+00
      1 Linear |R| = 7.216034e+00
      2 Linear |R| = 7.216034e+00
      3 Linear |R| = 7.216034e+00
      4 Linear |R| = 7.216034e+00
      5 Linear |R| = 7.216034e+00
      6 Linear |R| = 7.216034e+00
      7 Linear |R| = 7.216034e+00
      8 Linear |R| = 7.216034e+00
      9 Linear |R| = 7.216034e+00
     10 Linear |R| = 7.216034e+00
     11 Linear |R| = 7.216034e+00
     12 Linear |R| = 7.216034e+00
     13 Linear |R| = 7.216034e+00
     14 Linear |R| = 7.216034e+00
     15 Linear |R| = 7.216034e+00
     16 Linear |R| = 7.216034e+00
     17 Linear |R| = 7.216034e+00
     18 Linear |R| = 7.216034e+00
     19 Linear |R| = 7.216034e+00
     20 Linear |R| = 7.216034e+00
     21 Linear |R| = 7.216034e+00
     22 Linear |R| = 7.216034e+00
     23 Linear |R| = 7.216034e+00
     24 Linear |R| = 7.216034e+00
     25 Linear |R| = 7.216034e+00
     26 Linear |R| = 7.216034e+00
     27 Linear |R| = 7.216034e+00
     28 Linear |R| = 7.216033e+00
     29 Linear |R| = 7.216033e+00
     30 Linear |R| = 7.216032e+00
     31 Linear |R| = 7.216032e+00
     32 Linear |R| = 7.216032e+00
     33 Linear |R| = 7.216032e+00
     34 Linear |R| = 7.216032e+00
     35 Linear |R| = 7.216032e+00
     36 Linear |R| = 7.216032e+00
     37 Linear |R| = 7.216032e+00
     38 Linear |R| = 7.216032e+00
     39 Linear |R| = 7.216032e+00
     40 Linear |R| = 7.216032e+00
     41 Linear |R| = 7.216032e+00
     42 Linear |R| = 7.216032e+00
     43 Linear |R| = 7.216032e+00
     44 Linear |R| = 7.216032e+00
     45 Linear |R| = 7.216032e+00
     46 Linear |R| = 7.216032e+00
     47 Linear |R| = 7.216032e+00
     48 Linear |R| = 7.216032e+00
     49 Linear |R| = 7.216032e+00
     50 Linear |R| = 7.216032e+00
  Linear solve did not converge due to DIVERGED_ITS iterations 50
 1 Nonlinear |R| = 7.216032e+00
    |residual|_2 of individual variables:
                          q: 3.34388
                          p: 6.39449
      0 Linear |R| = 7.216032e+00
      1 Linear |R| = 7.216032e+00
      2 Linear |R| = 7.216032e+00
      3 Linear |R| = 7.216032e+00
      4 Linear |R| = 7.216032e+00
      5 Linear |R| = 7.216032e+00
      6 Linear |R| = 7.216032e+00
      7 Linear |R| = 7.216032e+00
      8 Linear |R| = 7.216032e+00
      9 Linear |R| = 7.216032e+00
     10 Linear |R| = 7.216032e+00
     11 Linear |R| = 7.216032e+00
     12 Linear |R| = 7.216032e+00
     13 Linear |R| = 7.216032e+00
     14 Linear |R| = 7.216032e+00
     15 Linear |R| = 7.216032e+00
     16 Linear |R| = 7.216032e+00
     17 Linear |R| = 7.216031e+00
     18 Linear |R| = 7.216031e+00
     19 Linear |R| = 7.216031e+00
     20 Linear |R| = 7.216030e+00
     21 Linear |R| = 7.216027e+00
     22 Linear |R| = 7.216022e+00
     23 Linear |R| = 7.216014e+00
     24 Linear |R| = 7.216005e+00
     25 Linear |R| = 7.215996e+00
     26 Linear |R| = 7.215991e+00
     27 Linear |R| = 7.215986e+00
     28 Linear |R| = 7.215985e+00
     29 Linear |R| = 7.215984e+00

**Reply by GiudGiud** [reply]
So initially you had them 'fully' coupled? with both solves in the same input file?
What you are showing here is two preconditioning blocks for individual solves, as in when using multiapps. 

What is the convergence log for ? For the fully coupled single-input? Or for another solve?

**Reply by Bearunner** [reply]
> So initially you had them 'fully' coupled? with both solves in the same input file?

Yes, they were 'fully' coupled with both solves in the one input file.

> What you are showing here is two preconditioning blocks for individual solves, as in when using multiapps.

Yes. They worked well both in their standalone apps if there was no coupling. They also worked well in Picard interaction (i.e., multiapps) coupling.
So I tried these two [Preconditioning]s in the 'fully' coupled with both solves in the one input file. But all failed.

> What is the convergence log for ? For the fully coupled single-input? Or for another solve?

### **Convergence log of 'fully' coupled with both solves in the one input file.**  It looks like the [Preconditioning] did not work well.

0 Nonlinear |R| = 7.216034e+00
|residual|_2 of individual variables:
q: 3.34388
p: 6.3945
0 Linear |R| = 7.216034e+00
1 Linear |R| = 7.216034e+00
2 Linear |R| = 7.216034e+00
3 Linear |R| = 7.216034e+00
4 Linear |R| = 7.216034e+00
5 Linear |R| = 7.216034e+00
6 Linear |R| = 7.216034e+00
7 Linear |R| = 7.216034e+00
8 Linear |R| = 7.216034e+00
9 Linear |R| = 7.216034e+00
10 Linear |R| = 7.216034e+00
11 Linear |R| = 7.216034e+00
12 Linear |R| = 7.216034e+00
13 Linear |R| = 7.216034e+00
14 Linear |R| = 7.216034e+00
15 Linear |R| = 7.216034e+00
16 Linear |R| = 7.216034e+00
17 Linear |R| = 7.216034e+00
18 Linear |R| = 7.216034e+00
19 Linear |R| = 7.216034e+00
20 Linear |R| = 7.216034e+00
21 Linear |R| = 7.216034e+00
22 Linear |R| = 7.216034e+00
23 Linear |R| = 7.216034e+00
24 Linear |R| = 7.216034e+00
25 Linear |R| = 7.216034e+00
26 Linear |R| = 7.216034e+00
27 Linear |R| = 7.216034e+00
28 Linear |R| = 7.216033e+00
29 Linear |R| = 7.216033e+00
30 Linear |R| = 7.216032e+00
31 Linear |R| = 7.216032e+00
32 Linear |R| = 7.216032e+00
33 Linear |R| = 7.216032e+00
34 Linear |R| = 7.216032e+00
35 Linear |R| = 7.216032e+00
36 Linear |R| = 7.216032e+00
37 Linear |R| = 7.216032e+00
38 Linear |R| = 7.216032e+00
39 Linear |R| = 7.216032e+00
40 Linear |R| = 7.216032e+00
41 Linear |R| = 7.216032e+00
42 Linear |R| = 7.216032e+00
43 Linear |R| = 7.216032e+00
44 Linear |R| = 7.216032e+00
45 Linear |R| = 7.216032e+00
46 Linear |R| = 7.216032e+00
47 Linear |R| = 7.216032e+00
48 Linear |R| = 7.216032e+00
49 Linear |R| = 7.216032e+00
50 Linear |R| = 7.216032e+00
Linear solve did not converge due to DIVERGED_ITS iterations 50
1 Nonlinear |R| = 7.216032e+00
|residual|_2 of individual variables:
q: 3.34388
p: 6.39449
0 Linear |R| = 7.216032e+00
1 Linear |R| = 7.216032e+00
2 Linear |R| = 7.216032e+00
3 Linear |R| = 7.216032e+00
4 Linear |R| = 7.216032e+00
5 Linear |R| = 7.216032e+00
6 Linear |R| = 7.216032e+00
7 Linear |R| = 7.216032e+00
8 Linear |R| = 7.216032e+00
9 Linear |R| = 7.216032e+00
10 Linear |R| = 7.216032e+00
11 Linear |R| = 7.216032e+00
12 Linear |R| = 7.216032e+00
13 Linear |R| = 7.216032e+00
14 Linear |R| = 7.216032e+00
15 Linear |R| = 7.216032e+00
16 Linear |R| = 7.216032e+00
17 Linear |R| = 7.216031e+00
18 Linear |R| = 7.216031e+00
19 Linear |R| = 7.216031e+00
20 Linear |R| = 7.216030e+00
21 Linear |R| = 7.216027e+00
22 Linear |R| = 7.216022e+00
23 Linear |R| = 7.216014e+00
24 Linear |R| = 7.216005e+00
25 Linear |R| = 7.215996e+00
26 Linear |R| = 7.215991e+00
27 Linear |R| = 7.215986e+00
28 Linear |R| = 7.215985e+00
29 Linear |R| = 7.215984e+00


### Convergence log of 'Picard iteration/MultiApps' coupled with both solves in their standalone input file


�[36mpipe_sub_inj0: �[39m
�[36mpipe_sub_inj0: �[39mTime Step 30, time = 3227.17, dt = 646.235
�[36mpipe_sub_inj0: �[39m 0 Nonlinear |R| = �[32m1.393859e-01�[39m
�[36mpipe_sub_inj0: �[39m 1 Nonlinear |R| = �[32m1.626299e-04�[39m
�[36mpipe_sub_inj0: �[39m 2 Nonlinear |R| = �[32m1.105107e-04�[39m
�[36mpipe_sub_inj0: �[39m 3 Nonlinear |R| = �[32m1.040351e-07�[39m
�[36mpipe_sub_inj0: �[39m 4 Nonlinear |R| = �[32m1.076632e-09�[39m
�[36mpipe_sub_inj0: �[39m�[32m Solve Converged!�[39m

�[35mFixed point residual norm after TIMESTEP_BEGIN MultiApps: �[32m1.405005e-08�[39m
�[35m
Main app solve:�[39m
 0 Nonlinear |R| = �[32m1.405005e-08�[39m
 1 Nonlinear |R| = �[32m4.987454e-11�[39m
�[32m Solve Converged!�[39m

 0 Picard |R| = �[32m1.591909e+00�[39m
 1 Picard |R| = �[33m1.591663e+00�[39m
 2 Picard |R| = �[32m4.275689e-01�[39m
 3 Picard |R| = �[32m2.974149e-01�[39m
 4 Picard |R| = �[32m2.178361e-01�[39m
 5 Picard |R| = �[32m1.633035e-01�[39m
 6 Picard |R| = �[32m1.236165e-01�[39m
 7 Picard |R| = �[32m9.395439e-02�[39m
 8 Picard |R| = �[32m7.153569e-02�[39m
 9 Picard |R| = �[32m5.450221e-02�[39m
10 Picard |R| = �[32m4.154714e-02�[39m
11 Picard |R| = �[32m3.167794e-02�[39m
12 Picard |R| = �[32m2.415499e-02�[39m
13 Picard |R| = �[32m1.842036e-02�[39m
14 Picard |R| = �[32m1.404766e-02�[39m
15 Picard |R| = �[32m1.071319e-02�[39m
16 Picard |R| = �[32m8.201140e-03�[39m
17 Picard |R| = �[32m6.254511e-03�[39m
18 Picard |R| = �[32m4.769999e-03�[39m
19 Picard |R| = �[32m3.637849e-03�[39m
20 Picard |R| = �[32m2.774418e-03�[39m
21 Picard |R| = �[32m2.115921e-03�[39m
22 Picard |R| = �[32m1.613717e-03�[39m
23 Picard |R| = �[32m1.230709e-03�[39m
24 Picard |R| = �[32m9.386061e-04�[39m
25 Picard |R| = �[32m7.158326e-04�[39m
26 Picard |R| = �[32m5.459334e-04�[39m
27 Picard |R| = �[32m4.163589e-04�[39m
28 Picard |R| = �[32m3.175383e-04�[39m
29 Picard |R| = �[32m2.421722e-04�[39m
30 Picard |R| = �[32m1.846938e-04�[39m
31 Picard |R| = �[32m1.408577e-04�[39m
32 Picard |R| = �[32m1.074259e-04�[39m
33 Picard |R| = �[32m8.192889e-05�[39m
34 Picard |R| = �[32m6.248350e-05�[39m
35 Picard |R| = �[32m4.765336e-05�[39m
36 Picard |R| = �[32m3.634308e-05�[39m
37 Picard |R| = �[32m2.771724e-05�[39m
38 Picard |R| = �[32m2.113879e-05�[39m
39 Picard |R| = �[32m1.612162e-05�[39m
40 Picard |R| = �[32m1.229524e-05�[39m
41 Picard |R| = �[32m9.377037e-06�[39m
42 Picard |R| = �[32m7.151450e-06�[39m
43 Picard |R| = �[32m5.454094e-06�[39m
44 Picard |R| = �[32m4.159596e-06�[39m
45 Picard |R| = �[32m3.172339e-06�[39m
46 Picard |R| = �[32m2.419403e-06�[39m
47 Picard |R| = �[32m1.844672e-06�[39m
48 Picard |R| = �[32m1.406827e-06�[39m
49 Picard |R| = �[32m1.072903e-06�[39m
50 Picard |R| = �[32m8.182389e-07�[39m
51 Picard |R| = �[32m6.240221e-07�[39m
52 Picard |R| = �[32m4.759045e-07�[39m
53 Picard |R| = �[32m3.629440e-07�[39m
54 Picard |R| = �[32m2.767958e-07�[39m
55 Picard |R| = �[32m2.110956e-07�[39m
56 Picard |R| = �[32m1.609900e-07�[39m
57 Picard |R| = �[32m1.227775e-07�[39m
58 Picard |R| = �[32m9.363509e-08�[39m
59 Picard |R| = �[32m7.140995e-08�[39m
60 Picard |R| = �[32m5.446004e-08�[39m
61 Picard |R| = �[32m4.153347e-08�[39m
62 Picard |R| = �[32m3.167511e-08�[39m
63 Picard |R| = �[32m2.415671e-08�[39m
64 Picard |R| = �[32m1.842289e-08�[39m
65 Picard |R| = �[32m1.405005e-08�[39m

**Reply by GiudGiud** [reply]
Do you still have the fully coupled input file?
and try this
```
[Preconditioning]
[basic]
type = SMP
full = true
petsc_options_iname = '-pc_type -pc_factor_shift_type '
petsc_options_value = '  lu NONZERO'
[]
[]
```
or some of these that are tuned for porous flow but maybe will work for porous flow + another phyiscs
https://mooseframework.inl.gov/modules/porous_flow/solvers.html
